resolvers/node/resolution.rs

```diff
@@ -1819,6 +1837,31 @@ impl<
     }
   }
 
+  pub(crate) fn legacy_fallback_resolve<'a>(
+    &self,
+    package_json: &'a PackageJson,
+    node_module_kind: NodeModuleKind,
+  ) -> Option<&'a str> {
+    if self.resolution_config.bundle_mode {
+      let maybe_browser = if self.resolution_config.prefer_browser_field {
+        package_json.browser.as_deref()
+      } else {
+        None
+      };
+      maybe_browser
+        .or(package_json.module.as_deref())
+        .or(package_json.main.as_deref())
+    } else {
+      match node_module_kind {
+        NodeModuleKind::Esm => package_json
+          .module
```

Comment:
I don't think we should be using `module` here as Node.js doesn't respect this. I wonder how many packages acctually hit this in the past as it seems we only used to do this when the `type` was "module"? I think we should just have this use `package_json.main.as_deref()` here. If anything, probably better to revert this as a fix and then see what breaks.